app/src/main/java/com/example/wnews/views/auth/AuthPresenter.kt

```diff
@@ -0,0 +1,87 @@
+package com.example.wnews.views.auth
+
+import android.content.SharedPreferences
+import android.util.Log
+import com.example.wnews.models.User
+import com.example.wnews.models.UserAuth
+import com.example.wnews.services.RetrofitProvider
+import okhttp3.Headers
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+
+class AuthPresenter(var sharedPref: SharedPreferences,val view: AuthView?){
+
+
+
+    fun onResponseLogIn(user: User){
+
+        val call: Call<UserAuth?>? = RetrofitProvider.getAuthService().createUser(user.email,user.password)
+        call!!.enqueue(object : Callback<UserAuth?> {
+
+            override fun onResponse(
+                call: Call<UserAuth?>?,
+                response: Response<UserAuth?>?
+            ) {
+
+                if (response!!.isSuccessful) {
+
+                    val headerList: Headers = response.headers()
+
+                    val userAuth = UserAuth()
+                    userAuth.apply {
+                        token = headerList["access-token"].toString()
+                        client = headerList["client"].toString()
+                        uid = user.email
+                    }
+
+                    saveUserAuth(userAuth)
+
+                    view!!.onAuthResponse(true)
+
+                } else if (response.errorBody() != null) {
+
+                    view!!.onAuthResponse(false)
+
+                }
+
+            }
+
+            override fun onFailure(call: Call<UserAuth?>?, t: Throwable?) {
```

Comment:
We should handle the `onFailure` case as well, as it may fail to make the API call (due to a bug, connection issues, backend issues, etc.). If this is the case, we should let the `View` know that there was an issue when performing the request, so it can display for example an error message to the user stating `Ups... something went wrong` or something like that.